examples/diff.c

```diff
@@ -86,49 +91,81 @@ int lg2_diff(git_repository *repo, int argc, char *argv[])
 	 *  * &lt;sha1&gt;
 	 *  * --cached
 	 *  * --nocache (don't use index data in diff at all)
+	 *  * --no-index &lt;file1&gt; &lt;file2&gt;
 	 *  * nothing
 	 *
 	 * Currently ranged arguments like &lt;sha1&gt;..&lt;sha2&gt; and &lt;sha1&gt;...&lt;sha2&gt;
 	 * are not supported in this example
 	 */
 
-	if (o.treeish1)
-		treeish_to_tree(&t1, repo, o.treeish1);
-	if (o.treeish2)
-		treeish_to_tree(&t2, repo, o.treeish2);
+	if (o.no_index >= 0) {
```

Comment:
This function is getting rather complex. I'd be very grateful if you'd split it up into multiple functions that compute a diff. E.g. something like:

```
if (o.so_index >= 0)
    compute_diff_no_index(&patch);
else if (o.cache == CACHE_NONE)
    compute_diff_no_cache(&patch);
else
    ...
```

Otherwise it's really hard to review this, even though it's not your fault :(

Reply:
Resolved for new no-index, not sure about rest of the code